linodecli/plugins/k8s-alpha.py

```diff
@@ -0,0 +1,286 @@
+
+    # We're done! We have merged the user's kubeconfigs.
+    # So, the user should be able to run something like
+    # `kubectl get pods --all-namespaces`
+    # and see the Linode CSI, CCM, and ExternalDNS controllers
+
+def delete(args, context):
+    parser = argparse.ArgumentParser("{} create".format(plugin_name), add_help=True)
+    parser.add_argument('name', metavar='NAME', type=str,
+                        help="A name for the cluster.")
+    parsed, _ = parser.parse_known_args(args)
+
+    # Get the appdir path
+    appdir = safe_make_appdir("k8s-alpha")
+    terrapath = os.path.join(appdir, parsed.name)
+    # Move to the terraform directory
+    os.chdir(terrapath)
+    call_or_exit(['terraform', 'destroy'])
+
+    # TODO: Also delete all NodeBalancers and Volumes using the cluster prefix
+
+def quoted_string_or_bare_int(val):
+    if type(val) is int:
+        return val
+    elif type(val) is str:
+        return '"{}"'.format(val)
+    else:
+        return None
+
+def gen_terraform_file(context, cluster_name, hashname):
+    tf_file_parts = []
+
+    for varname in tf_var_map.keys():
+        tf_file_parts.append("""variable "{tf_varname}" {{
+  default = {default}
+}}
+""".format(tf_varname=tf_var_map[varname]['name'],
+           default=quoted_string_or_bare_int(tf_var_map[varname]['default'])))
+    
+    tf_file_parts.append("""module "k8s" {{
+  source  = "git@github.com:linode/terraform-linode-k8s.git?ref=for-cli"
+
+  linode_token = "{token}"
+  
+  linode_group = "ka{hashname}-{cluster_name}"
+""".format(token=context.token,
+           cluster_name=cluster_name,
+           hashname=hashname,))
+
+    for varname in tf_var_map.keys():
+        tf_file_parts.append("""
+  {tf_varname} = "${{var.{tf_varname}}}"
+""".format(tf_varname=tf_var_map[varname]['name']))
+
+    tf_file_parts.append("}\n")
+
+    return ''.join(tf_file_parts)
+
+def gen_terraform_args(parsed):
+    args = []
+    for varname in tf_var_map.keys():
+        args = args + ['-var', "{}={}".format(tf_var_map[varname]['name'], getattr(parsed, varname))]
+    return args
+
+def call_or_exit(*args, **kwargs):
+    ret = spcall(*args, **kwargs)
+    if ret != 0:
+        sys.exit(ret)
+
+def replace_kubeconfig_user(terrapath, cluster_name):
+    """
+    If we leave the user as kubernetes-admin. Then the configs don't flatten properly.
+    All of them try to create creds for kubernetes-admin.
+    """
+    kubeconfig_fp = os.path.join(terrapath, "{}.conf".format(cluster_name))
+    kubeconfig = open(kubeconfig_fp).read()
+    kubeconfig = kubeconfig.replace('kubernetes-admin', cluster_name)
+
+    kubeconfig_new_fp = os.path.join(terrapath, "{}_new.conf".format(cluster_name))
+    kubeconfig_new = open(kubeconfig_new_fp, 'w')
+    kubeconfig_new.write(kubeconfig)
+
+    return kubeconfig_new_fp
+
+def safe_make_appdir(appname):
+    if sys.platform == 'win32':
+        appdir = os.path.join(os.environ['APPDATA'], appname)
```

Comment:
I'm not very familiar with Windows these days, but the internet suggests this will return `AppData\Roaming`.  Either way, `os.environ.get('APPDATA')` won't raise if it isn't set (although I'm not sure what good behavior would be if it wasn't set)